playwright/containers/n8n-test-container-creation.ts

```diff
@@ -0,0 +1,322 @@
+/**
+ * n8n Test Containers Setup
+ * This file provides a complete n8n container stack for testing with support for:
+ * - Single instances (SQLite or PostgreSQL)
+ * - Queue mode with Redis
+ * - Multi-main instances with nginx load balancing
+ * - Parallel execution (multiple stacks running simultaneously)
+ *
+ * Key features for parallel execution:
+ * - Dynamic port allocation to avoid conflicts (handled by testcontainers)
+ * - WebSocket support through nginx load balancer
+ */
+
+import type { StartedNetwork, StartedTestContainer } from 'testcontainers';
+import { GenericContainer, Network, Wait } from 'testcontainers';
+
+import {
+	setupNginxLoadBalancer,
+	setupPostgres,
+	setupRedis,
+} from './n8n-test-container-dependencies';
+import { TestError } from '../Types';
+
+// --- Constants ---
+
+const POSTGRES_IMAGE = 'postgres:16-alpine';
+const REDIS_IMAGE = 'redis:7-alpine';
+const NGINX_IMAGE = 'nginx:stable';
+const N8N_E2E_IMAGE = 'n8n-local:dev';
+
+// Default n8n image (can be overridden via N8N_DOCKER_IMAGE env var)
+const N8N_IMAGE = process.env.N8N_DOCKER_IMAGE ?? N8N_E2E_IMAGE;
```

Comment:
Maybe use `||` incase `N8N_DOCKER_IMAGE` happens to be an empty string

```suggestion
const N8N_IMAGE = process.env.N8N_DOCKER_IMAGE || N8N_E2E_IMAGE;
```